keps/sig-scheduling/20191104-ready-pod-priority.md

```diff
@@ -0,0 +1,129 @@
+  - 
+reviewers:
+  - TBD
+approvers:
+  - TBD
+editor: 
+  - TBD
+creation-date: yyyy-mm-dd
+last-updated: yyyy-mm-dd
+status: provisional
+see-also:
+  - 
+replaces:
+  - 
+superseded-by:
+  - 
+---
+
+# Ready Pod Priority
+
+## Table of Contents
+
+<!-- toc -->
+- [Release Signoff Checklist](#release-signoff-checklist)
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [User Stories [optional]](#user-stories-optional)
+    - [Story 1](#story-1)
+    - [Story 2](#story-2)
+  - [Implementation Details/Notes/Constraints [optional]](#implementation-detailsnotesconstraints-optional)
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Design Details](#design-details)
+  - [Test Plan](#test-plan)
+  - [Graduation Criteria](#graduation-criteria)
+    - [Examples](#examples)
+      - [Alpha -&gt; Beta Graduation](#alpha---beta-graduation)
+      - [Beta -&gt; GA Graduation](#beta---ga-graduation)
+      - [Removing a deprecated flag](#removing-a-deprecated-flag)
+  - [Upgrade / Downgrade Strategy](#upgrade--downgrade-strategy)
+  - [Version Skew Strategy](#version-skew-strategy)
+- [Implementation History](#implementation-history)
+- [Drawbacks [optional]](#drawbacks-optional)
+- [Alternatives [optional]](#alternatives-optional)
+- [Infrastructure Needed [optional]](#infrastructure-needed-optional)
+<!-- /toc -->
+
+## Release Signoff Checklist
+
+- [ ] kubernetes/enhancements issue in release milestone, which links to KEP (this should be a link to the KEP location in kubernetes/enhancements, not the initial KEP PR)
+- [ ] KEP approvers have set the KEP status to `implementable`
+- [ ] Design details are appropriately documented
+- [ ] Test plan is in place, giving consideration to SIG Architecture and SIG Testing input
+- [ ] Graduation criteria is in place
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://github.com/kubernetes/enhancements/issues
+[kubernetes/kubernetes]: https://github.com/kubernetes/kubernetes
+[kubernetes/website]: https://github.com/kubernetes/website
+
+## Summary
+
+New scheduler priority `ReadyPodPriority` which adds a score to prefer nodes with least non ready pods.
+
+## Motivation
+
+In our environment we have pods that take a while to pass ready/liveness checks as these pods have caches which need to backfill to memory prior to being ready, during this time memory and cpu resources are spiked. 
+By adding a score to favor nodes with less non ready pods, this helped us by preventing deployments to overload a node, especially a newly entered node which is idle, from getting overloaded.
```

Comment:
Overloaded is mainly characterized by excessive use of cpu, memory or other resources. If we can perceive whether the actual load of the current resource can cover this case. My point is that we should focus directly on the load situation rather than the factors that cause the load to be high, because there are many such factors.

Reply:
Thanks @wgliang for the comment, I've updated the motivation section to focus on load as suggested. Please let me know it the update is better described and makes sense.